### PR TITLE
Modify socket catalog

### LINKED ISSUE #9: Update auction page with actual catalog details
As of now auction page shows just a number on which people bid. Change that with actual catalog details by fetching it from backend. This requires a bit of change to socket code too.

### PR DESCRIPTION
**Pull request by shakti97**
This issue will solve and close #9 and #14 issue

### CODE REVIEW

File: assets/app/jsx/AdminPanel.jsx [hunk 1]
```diff
@@ -108,7 +111,25 @@ class AdminPanel extends Component {
             .catch(err => {
                 console.log('Some error occured');
             });
-
+        dataFetch('/getCatalog', data)
+            .then(response => {
+                if (response.status_code == 200) {
+                    console.log(response);
```
Comment (hariprasanths) [inline assets/app/jsx/AdminPanel.jsx:117]: Remove debug logs

File: assets/app/jsx/AdminPanel.jsx [hunk 2]
```diff
@@ -108,7 +111,25 @@ class AdminPanel extends Component {
...
+                    this.setState(
+                        {
+                            catalogs: response.message
+                        },
+                        () => {
+                            console.log('this.state.catalogs ', this.state.catalogs);
+                        }
+                    );
+                } else {
+                    console.log('Error in fetching catalog');
```
Comment (hariprasanths) [inline assets/app/jsx/AdminPanel.jsx:127]: Give the error feedback to the user
Show the error message sent from the backend

File: assets/app/jsx/AdminPanel.jsx [hunk 3]
```diff
@@ -108,7 +111,25 @@ class AdminPanel extends Component {
...
+                }
+            })
+            .catch(err => {
+                console.log('Some Error Occured');
```
Comment (hariprasanths) [inline assets/app/jsx/AdminPanel.jsx:131]: Show err.message

File: assets/app/jsx/AdminPanel.jsx [hunk 4]
```diff
@@ -146,8 +167,71 @@ class AdminPanel extends Component {
...
+                    console.log('bidDetails ', bidDetails);
+                    this.setState(
+                        {
+                            bidDetails
+                        },
+                        () => {
+                            const {currentBid: final_price, bidHolderId: user_id} = this.state.bidDetails;
+                            data.final_price = final_price;
+                            data.user_id = user_id;
+                            dataFetch('/saveAuctionSummary', data)
```
Comment (hariprasanths) [inline assets/app/jsx/AdminPanel.jsx:198]: Multiple entries gets added in auction summary for a single item:
2 entries for 2nd item sold
3 for 3rd item sold and so on

File: assets/app/jsx/AdminPanel.jsx [hunk 5]
```diff
@@ -245,6 +329,44 @@ class AdminPanel extends Component {
...
+                                                        disabled={start && start !== catalog.id}
+                                                        onClick={() => this.markBiddingStart(catalog.id)}>
+                                                        Start
+                                                    </button>
+                                                )}
+                                                <button
+                                                    className="btn btn-danger m-1"
+                                                    disabled={start && start !== catalog.id}
+                                                    onClick={() => this.markSold(catalog.id)}>
+                                                    Sold
```
Comment (hariprasanths) [inline assets/app/jsx/AdminPanel.jsx:361]: sold button in admin panel should only br available after starting the bid

**Review by hariprasanths — CHANGES_REQUESTED** [review]

File: assets/app/jsx/AdminPanel.jsx (hunk 5 above)
Comment (shakti97) [reply assets/app/jsx/AdminPanel.jsx:361]: Done

File: assets/app/jsx/AdminPanel.jsx (hunk 4 above)
Comment (shakti97) [reply assets/app/jsx/AdminPanel.jsx:198]: Resolved

File: assets/app/jsx/AdminPanel.jsx (hunk 3 above)
Comment (shakti97) [reply assets/app/jsx/AdminPanel.jsx:131]: Resolved

File: assets/app/jsx/AdminPanel.jsx (hunk 2 above)
Comment (shakti97) [reply assets/app/jsx/AdminPanel.jsx:127]: Resolved

File: assets/app/jsx/AdminPanel.jsx (hunk 1 above)
Comment (shakti97) [reply assets/app/jsx/AdminPanel.jsx:117]: Resolved

**Comment by shakti97** [comment]
Close this , placed another pr( Catalogs ) which includes the recommended changes.

**Pull request closed without merging**